lib/puppet/util/servicenow.rb

```diff
@@ -151,9 +151,13 @@ def calculate_satisfied_conditions(report_status, resource_statuses, incident_cr
   module_function :calculate_satisfied_conditions
 
   def incident_description(satisfied_conditions, settings_hash)
+    facts = Puppet::Node::Facts.indirection.find(host).values
+    included_facts = settings_hash['include_facts'].map { |name| "#{name}: #{JSON.pretty_generate(facts.dig(*name.split('.')))}" }
+
     'This incident was created based on the following conditions: '\
     "#{satisfied_conditions.join(', ')}. See the PE console for the full report. "\
-    "You can access the PE console at #{settings_hash['pe_console_url']}."
+    "You can access the PE console at #{settings_hash['pe_console_url']}."\
+    "\n\nfacts:\n\n#{included_facts.join("\n")}"
```

Comment:
This seems fine. In fact, some of the code here could actually be useful for event management. For example, we could set the event's `node` field to the `fqdn` fact and ServiceNow will automatically bind the generated alert to the CMDB item with that matching `fqdn` (see https://docs.servicenow.com/bundle/orlando-it-operations-management/page/product/event-management/concept/c_EMIntegrateRequirementEvent.html or the PIE-329 doc).

Why add the `include_facts` parameter though? If it's to limit noise, is the noise really that bad? Asking b/c I can imagine a frustrated admin wondering why an important fact doesn't show up in the incident only to find out that they forgot to include it in the list. Thus if the noise isn't bad, I say we dump everything until someone complains about it.

Reply:
The noise would not be bad for event management because in that case the intended audience to some extent is the machine learning algorithms and grouping algorithms, and in that case, the more data the better. But for human consumption the noise makes it really difficult to easily see what you're looking for. I guess it's not inconceivable that some poor soul might want to hurt themselves in that way though, so maybe an `all` keyword that just includes everything might be in order.

Reply:
Hmm, couldn't they just ctrl+F? Asking b/c I can also imagine a scenario where someone's wondering why fact X isn't included only to find (frustratingly) that someone else may not have added it to `include_fact`. Just feels like something that should be added after customer feedback, but I also don't feel strongly about that at all so yeah, at the very least some kind of `all` keyword would be useful.

Reply:
They could, but, we have to keep in mind that most users don't know what ctrl+f is. Most users have no idea you can search a web page for text in a browser and most, even a lot of IT Pros, don't know that keyboard shortcuts are even a thing.

Reply:
Huh, I didn't know that. Thanks for the useful info.